01_deployment.tf

```diff
@@ -68,6 +68,17 @@ resource "kubernetes_deployment_v1" "deployment" {
         }
         share_process_namespace = var.hostConfig.shareProcessNamespace
 
+        dynamic "toleration" {
+          for_each = var.podResourceTypeConfig.toleration
```

Comment:
as per definition the tolerations are of type list.

Keep in mind that for_each loops for lists will order by index, so changes to the tolerations list would probably generate quite a big diff.

tbh I don't think this will change too often, so I don't see it necessarily as a problem. Just wanted to clarify that you thought about the implications that this will have. Feel free to keep it or change it to  a map, so that ordering doesn't matter any longer. I'll accept it either way

Reply:
I changed it into a map.